filemutex_flock.go

```diff
@@ -19,7 +25,19 @@ type FileMutex struct {
 }
 
 func New(filename string) (*FileMutex, error) {
-	fd, err := unix.Open(filename, unix.O_CREAT|unix.O_RDONLY, mkdirPerm)
+	return new(filename, mkdirPerm)
+}
+
+func NewWithUnixFilePerm(filename string, perm uint32) (*FileMutex, error) {
```

Comment:
Can you make `perm` be of type `os.FileMode` and name this `NewWithMode`?

Reply:
sure